corehq/apps/reports/models.py

```diff
@@ -566,15 +566,12 @@ def delete_user(self, id):
 
 class QueryStringHash(models.Model):
     date_created = models.DateTimeField(auto_now_add=True)
+    domain = models.CharField(max_length=64)
     last_accessed = models.DateTimeField(auto_now=True)
-    query_hash = models.CharField(max_length=64, db_index=True)
+    query_id = models.CharField(max_length=32, db_index=True, unique=True)
```

Comment:
nit: Could use `UUIDField` for more efficient storage and indexes.

Reply:
Is it supposed to be possible to filter by UUIDField using the string version of the UUID? It doesn't feel like it should work but it does when testing locally so I'm conflicted on whether to implement this or not.

Reply:
nevermind! I can't find a clear answer in the docs but [this answer](https://stackoverflow.com/questions/72996640/uuidfield-vs-charfield-with-a-uuid-in-it) claims UUIDField can parse strings as valid (or not) UUIDs